btk/utils.py

```diff
@@ -1,489 +0,0 @@
-    return blend_catalog
-
-
-def group_sampling_function(Args, catalog):
-    """Blends are defined from *groups* of galaxies from the CatSim
-    catalog previously analyzed with WLD.
-
-    The group is centered on the middle of the postage stamp.
-    Function only draws galaxies that lie within the postage stamp size
-    determined in Args.
-
-    Note: the pre-run WLD images are not used here. We only use the pre-run
-    catalog (in i band) to identify galaxies that belong to a group.
-    """
-    if not hasattr(Args, 'wld_catalog_name'):
-        raise Exception("A pre-run WLD catalog  name should be input as "
-                        "Args.wld_catalog_name")
-    else:
-        wld_catalog = astropy.table.Table.read(Args.wld_catalog_name,
-                                               format='fits')
-    # randomly sample a group.
-    group_ids = np.unique(wld_catalog['grp_id'][wld_catalog['grp_size'] >= 2])
-    group_id = np.random.choice(group_ids, replace=False)
-    # get all galaxies belonging to the group.
-    ids = wld_catalog['db_id'][wld_catalog['grp_id'] == group_id]
-    blend_catalog = astropy.table.vstack(
-        [catalog[catalog['galtileid'] == i] for i in ids])
-    # Set mean x and y coordinates of the group galaxies to the center of the
-    # postage stamp.
-    blend_catalog['ra'] -= np.mean(blend_catalog['ra'])
-    blend_catalog['dec'] -= np.mean(blend_catalog['dec'])
-    # convert ra dec from degrees to arcsec
-    blend_catalog['ra'] *= 3600
-    blend_catalog['dec'] *= 3600
-    # Add small random shift so that center does not perfectly align with
-    # the stamp center
-    dx, dy = btk.create_blend_generator.get_random_center_shift(
-        Args, 1, maxshift=3 * Args.pixel_scale)
-    blend_catalog['ra'] += dx
-    blend_catalog['dec'] += dy
-    # make sure galaxy centers don't lie too close to edge
-    cond1 = np.abs(blend_catalog['ra']) < Args.stamp_size / 2. - 3
-    cond2 = np.abs(blend_catalog['dec']) < Args.stamp_size / 2. - 3
-    no_boundary = blend_catalog[cond1 & cond2]
-    if len(no_boundary) == 0:
-        return no_boundary
-    # make sure number of galaxies in blend is less than Args.max_number
-    # randomly select max_number of objects if larger.
-    num = min([len(no_boundary), Args.max_number])
-    select = np.random.choice(range(len(no_boundary)), num, replace=False)
-    return no_boundary[select]
-
-
-class Basic_measure_params(measure.Measurement_params):
-    """Class to perform detection and deblending with SEP"""
-
-    def get_centers(self, image):
-        """Return centers detected when object detection and photometry
-        is done on input image with SEP.
-        Args:
-            image: Image (single band) of galaxy to perform measurement on.
-
-        Returns:
-                centers: x and y coordinates of detected  centroids
-        """
-        # set detection threshold to 5 times std of image
-        threshold = 5*np.std(image)
```

Comment:
That is going to give a rather high threshold. Setting to 5*std(noise) is usually a good threshold but your choice will end up being slightly higher.